### PR TITLE
[Wait for #1381] [capi] Implement dataset_set_property_for_usage

### PR DESCRIPTION
**Pull request by zhoonit**
**[Pending commit: #1381]**

- [capi] Implement dataset_set_property_for_usage

```
This patch implements `ml_train_dataset_set_property_for_usage` and it's
test

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: api/capi/src/nntrainer.cpp [hunk 1]
```diff
@@ -764,50 +764,103 @@ int ml_train_dataset_create_with_file(ml_train_dataset_h *dataset,
...
-    }
+  status = ml_train_dataset_set_property_for_usage_(
+    dataset, ML_TRAIN_DATASET_DATA_USAGE_TEST, arg_list);
+  if (status != ML_ERROR_NONE || status != ML_ERROR_NOT_SUPPORTED) {
     return status;
-  };
-  status = nntrainer_exception_boundary(f);
+  }
 
   return status;
```
Comment (kparichay) [inline api/capi/src/nntrainer.cpp:842]: shouldn't this be ML_ERROR_NONE? as this can return ML_ERROR_NOT_SUPPORTED

Comment (zhoonit) [reply api/capi/src/nntrainer.cpp:842]: Yeah you are right. Nice catch :)

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**